htdocs/product/class/product.class.php

```diff
@@ -803,7 +803,7 @@ function update($id, $user, $notrigger=false, $action='update')
                     // End call triggers
 				}
 
-				if (! $error && (is_object($this->oldcopy) && $this->oldcopy->ref != $this->ref))
+				if (! $error && (is_object($this->oldcopy) && $this->oldcopy->ref !== $this->ref))
```

Comment:
Ref should be with of type string. Even if value inside is numeric.
What is field that has type int instead of string ?  The oldcopy->ref or ->ref ?